processor/elasticprocessor/processor.go

```diff
@@ -4,44 +4,61 @@ import (
 	"context"
 	"strings"
 
-	"github.com/tommyers-elastic/opentelemetry-collector-contrib/processor/elasticprocessor/internal/hostmetrics"
-
+	"github.com/elastic/opentelemetry-lib/remappers/hostmetrics"
+	"go.opentelemetry.io/collector/pdata/pcommon"
 	"go.opentelemetry.io/collector/pdata/plog"
 	"go.opentelemetry.io/collector/pdata/pmetric"
 	"go.opentelemetry.io/collector/pdata/ptrace"
 	"go.opentelemetry.io/collector/processor"
-
 	"go.uber.org/zap"
 )
 
+// remapper interface defines the Remap method that should be implemented by different remappers
+type remapper interface {
+	Remap(pmetric.ScopeMetrics, pmetric.MetricSlice, pcommon.Resource)
+}
+
 type ElasticProcessor struct {
-	cfg     *Config
-	logger  *zap.Logger
-	storage map[string]any
+	cfg       *Config
+	logger    *zap.Logger
+	storage   map[string]any
+	remappers []remapper
 }
 
 func newProcessor(set processor.CreateSettings, cfg *Config) *ElasticProcessor {
-	return &ElasticProcessor{cfg: cfg, logger: set.Logger, storage: make(map[string]any)}
+	remappers := []remapper{
+		hostmetrics.NewRemapper(set.Logger, hostmetrics.WithSystemIntegrationDataset(true)),
+	}
+	return &ElasticProcessor{
+		cfg:       cfg,
+		logger:    set.Logger,
+		storage:   make(map[string]any),
+		remappers: remappers,
+	}
 }
 
+// processMetrics processes the given metrics and applies remappers if configured.
 func (p *ElasticProcessor) processMetrics(_ context.Context, md pmetric.Metrics) (pmetric.Metrics, error) {
 	for i := 0; i < md.ResourceMetrics().Len(); i++ {
 		resourceMetric := md.ResourceMetrics().At(i)
 		rm := resourceMetric.Resource()
 
 		for j := 0; j < resourceMetric.ScopeMetrics().Len(); j++ {
 			scopeMetric := resourceMetric.ScopeMetrics().At(j)
-
+			// Apply remappers if AddSystemMetrics is enabled in the configuration
 			if p.cfg.AddSystemMetrics {
-				if strings.HasPrefix(scopeMetric.Scope().Name(), "otelcol/hostmetricsreceiver") {
-					if err := hostmetrics.AddElasticSystemMetrics(scopeMetric, rm, p.storage); err != nil {
-						p.logger.Error("error adding hostmetrics data", zap.Error(err))
+				if len(p.remappers) > 0 {
+					// Apply remappers only if the scope name has the prefix "otelcol/hostmetricsreceiver"
+					if strings.HasPrefix(scopeMetric.Scope().Name(), "otelcol/hostmetricsreceiver") {
+						for _, r := range p.remappers {
+							r.Remap(scopeMetric, scopeMetric.Metrics(), rm)
+						}
 					}
+
 				}
 			}
 		}
```

Comment:
Assuming we create the remappers based on the `AddSystemMetrics` flag. Additionally, checking for validity of the source metric should be left to the remappers i.e. remappers should remap only when the scope is `otelcol/hostmetricsreceiver` for our case (this is true: [code ref](https://github.com/elastic/opentelemetry-lib/blob/3234f90c8fca7c6304f750b960b21c9437fefc91/remappers/hostmetrics/hostmetrics.go#L81-L83)). So, we can rewrite this piece:

```go
		for j := 0; j < resourceMetric.ScopeMetrics().Len(); j++ {
			scopeMetric := resourceMetric.ScopeMetrics().At(j)
			for _, r := range p.remappers {
				r.Remap(scopeMetric, scopeMetric.Metrics(), rm)
			}
		}
```

Reply:
Yes, now if the AddSystemMetrics  check is placed up, we can rewrite this piece.